app/models/mixins/relationship_mixin.rb

```diff
@@ -565,6 +623,11 @@ def add_children(*child_objs)
   alias_method :add_child, :add_children
 
   def parent=(parent)
+    if use_ancestry?
+      call_ancestry_method(:parent=, parent)
+      return
+    end
```

Comment:
NOTE: this is one of the main wins of going from relationships to parents.
It is only possible to store one parent in the field - so the corruption we have seen with an object having multiple parents is no longer possible.